### PR TITLE
feat/TestRng: impl RngCore trait for TestRng

### PR DESCRIPTION
**Pull request by maqi**

### CODE REVIEW

File: src/rng.rs [hunk 1]
```diff
@@ -137,6 +137,24 @@ mod test {
...
+        fn next_u64(&mut self) -> u64 {
+            self.0.next_u64()
+        }
+
+        fn fill_bytes(&mut self, dest: &mut [u8]) {
+            self.0.fill_bytes(dest)
+        }
+
+        fn try_fill_bytes(&mut self, _dest: &mut [u8]) -> Result<(), rand_crypto::Error> {
+            Ok(())
```
Comment (jeanphilippeD) [inline src/rng.rs]: Probably need to call `self.0.fill_bytes(dest)` before returning Ok, this should not have stub non functional implementation.

Comment (dirvine) [reply src/rng.rs]: Does `self.0.fill_bytes(dest)` not return the correct Result? I mean should that not be the last line with no `;` ? @maqi

Comment (madadam) [reply src/rng.rs]: `fill_bytes` returns `()`, but we need to return `Result`.

Comment (maqi) [reply src/rng.rs]: We can do something like `Ok(self.0.fill_bytes(dest))`
but I think it's bit mis-leading?

Comment (jeanphilippeD) [reply src/rng.rs]: Yes, it would be misleading.

**Pull request merged**